docs: rework Buffer page (#1388)
nodejs/nodejs.dev: src/documentation/0051-node-buffers/index.md

console.log(buf[2]) //121
```

> Those numbers are the UTF-8 bytes that identifies the character in the buffer position (H => 72, e => 101, y => 121). This happens because `Buffer.from()` uses UTF-8 by default.
> Keep in mind that some characters may occupy more than one byte in the buffer (é => 195 169).

Reviewer: ```suggestion
Those numbers are the UTF-8 bytes that identify the characters in the buffer (`H` → `72`, `e` → `101`, `y` → `121`). This happens because `Buffer.from()` uses UTF-8 by default.
Keep in mind that some characters may occupy more than one byte in the buffer (`é` → `195 169`).
```

```js
const buf = Buffer.from('Hey!')
buf[1] = 111 //o in UTF-8
console.log(buf.toString()) //Hoy!
```

### Slice a buffer

If you want to create a partial visualization of a buffer, you can create a slice. A slice is not a copy: the original buffer is still the source of truth. If that changes, your slice changes.

Use the `subarray()` method to create it. The first parameter is the starting position, and you can specify an optional second parameter with the end position:

```js
const buf = Buffer.from('Hey!')
buf.subarray(0).toString() //Hey!
const slice = buf.subarray(0, 2)
console.log(slice.toString()) //He
buf[1] = 111 //o
console.log(slice.toString()) //Ho
```

### Copy a buffer

Copying a buffer is possible using the `set()` method:

```js
const buf = Buffer.from('Hey!')
let bufcopy = Buffer.alloc(4) //allocate 4 bytes
bufcopy.set(buf)
```

> By default you copy the whole buffer. If you only want to copy a part of the buffer, you can use `.subarray()` from above and the `offset` argument that specifies an offset to write to:

Reviewer: ```suggestion
By default you copy the whole buffer. If you only want to copy a part of the buffer, you can use `.subarray()` show above and the `offset` argument that specifies an offset to write to:
```

Author: I don’t quite understand this one, to be honest – `show above`?

Reviewer: Yeah, that was a typo 😆 fixed.

> By default you copy the whole buffer. If you only want to copy a part of the buffer, you can use `.subarray()` shown above and the `offset` argument that specifies an offset to write to:

Reviewer 2: Non-blocking, but FWIW, "above" and "below" are generally discouraged as making assumptions about the way the web page/documentation is being consumed. (Imagine, for example, a screen reader.) https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/a/above suggests _previous_, _preceding_, or _earlier_ as alternatives.  So maybe this?:

```suggestion
By default you copy the whole buffer. If you only want to copy a part of the buffer, you can use `.subarray()` shown earlier and the `offset` argument that specifies an offset to write to:
```

Or even remove it entirely?:

```suggestion
By default you copy the whole buffer. If you only want to copy a part of the buffer, you can use `.subarray()` and the `offset` argument that specifies an offset to write to:
```

Reviewer: I though of removing it but I wasn't sure.  I think removing sounds good 👍🏻
